src/Client.php

```diff
@@ -63,68 +69,75 @@ public function getFolders() : array
         return array_keys($this->hordeClient->listMailboxes('*'));
     }
 
-    /**
+     /**
      * Finds the emails matching the query. If $query is null, then it will fetch the emails in the inbox.
-     *
+     * @param Query $query Description
+     * @param boolean $peek sets the peek option, "false" fetched emails' state will be set to seen "true" no change of state default is true
      * @return Email[]
      */
-    public function getEmails(Query $query = null) : array
-    {
+    public function getEmails(Query $query = null, $peek = true): array {
         $hordeQuery = new Horde_Imap_Client_Search_Query();
 
         $query = $query ?: new Query;
 
         if ($query->getYoungerThan() !== null) {
             $hordeQuery->intervalSearch(
-                $query->getYoungerThan(),
-                Horde_Imap_Client_Search_Query::INTERVAL_YOUNGER
+                    $query->getYoungerThan(), Horde_Imap_Client_Search_Query::INTERVAL_YOUNGER
             );
         }
-
+        $this->setPeek($peek);
         $this->setFlags($hordeQuery, $query);
+
         return $this->searchAndFetch($query->getFolder(), $hordeQuery);
     }
 
     /**
      * Finds the email Ids matching the query. If $query is null, then it will fetch the email Ids in the inbox.
      *
      * This method is obviously more efficient than getEmails() if you want to synchronize local mails.
-     *
+     * 
+     * @param Query $query Description
+     * @param boolean $peek sets the peek option, "false" fetched emails' state will be set to seen "true" no change of state default is true
      * @return string[]
      */
-    public function getEmailIds(Query $query = null) : array
-    {
+    public function getEmailIds(Query $query = null, $peek = true): array {
         $hordeQuery = new Horde_Imap_Client_Search_Query();
 
         $query = $query ?: new Query;
 
         if ($query->getYoungerThan() !== null) {
             $hordeQuery->intervalSearch(
-                $query->getYoungerThan(),
-                Horde_Imap_Client_Search_Query::INTERVAL_YOUNGER
+                    $query->getYoungerThan(), Horde_Imap_Client_Search_Query::INTERVAL_YOUNGER
             );
         }
 
+        $this->setPeek($peek);
         $this->setFlags($hordeQuery, $query);
+
         return $this->search($query->getFolder(), $hordeQuery);
     }
 
     /**
+     * @param String $id Description
```

Comment:
This line should be removed (there is no additional information, the type-hint is enough)

Reply:
Sure